src/shell/Shell.cpp

```diff
@@ -795,6 +795,84 @@ static void executeWAST(Store* store, const std::string& filename, const std::ve
     }
 }
 
+static void runExports(Store* store, const std::string& filename, const std::vector<uint8_t>& src, std::string& exportToRun)
+{
+    auto parseResult = WASMParser::parseBinary(store, filename, src.data(), src.size());
+
+    if (!parseResult.second.empty()) {
+        fprintf(stderr, "parse error: %s\n", parseResult.second.c_str());
+        return;
+    }
+
+    auto module = parseResult.first;
+
+    const auto& importTypes = module->imports();
+
+    ExternVector importValues;
+
+    if (importTypes.size() != 0) {
+        fprintf(stderr, "error: module has imports, but imports are not supported\n");
+        return;
+    }
+
+    struct RunData {
+        Module* module;
+        ExternVector& importValues;
+        ExportType* exp;
+    } data = { module.value(), importValues, nullptr };
+    Walrus::Trap trap;
+
+    for (auto&& exp : module->exports()) {
+        if (exp->exportType() == ExportType::Function) {
+            if (exportToRun != exp->name() && exportToRun != "*") {
+                continue;
+            }
+
+            data.exp = exp;
+            trap.run([](ExecutionState& state, void* d) {
+                auto data = reinterpret_cast<RunData*>(d);
+                Instance* instance = data->module->instantiate(state, data->importValues);
+                auto fn = instance->function(data->exp->itemIndex());
+                FunctionType* fnType = fn->asDefinedFunction()->moduleFunction()->functionType();
+
+                if (!fnType->param().empty()) {
+                    printf("warning: function %s has params, but params are not supported\n", data->exp->name().c_str());
```

Comment:
`runExports` cannot support cases in which there are any import values or parameters required for export functions.
I'm wondering that `runExports` feature is really necessary although it only supports in limited cases.
Or is this feature commonly enabled in other WASM engines?

Reply:
This is a common feature of WASM engines. For example wasmer has it and WABT has it too.

Reply:
The primary reason is running benchmarks. And it is only shell modifications, so it should have no effects on the engine. We could add support for certain imports if it is necessary.

Reply:
I see. It would be also great to add and run benchmarks that need this feature in the future.